packages/components/pie-modal/src/modal.scss

```diff
@@ -1,51 +1,56 @@
-$modal-size-l: 1080px;
-$modal-size-m: 600px;
-$modal-size-s: 450px;
-
-$modal-border-radius: var(--dt-radius-rounded-d);
-$modal-font: var(--dt-font-interactive-m-family);
-$modal-bg-color: var(--dt-color-container-default);
-$modal-elevation: var(--dt-elevation-04);
-
-$modal-header-font-size: calc(var(--dt-font-heading-m-size--wide) * 1px);
-$modal-header-font-line-height: calc(var(--dt-font-heading-m-line-height--wide) * 1px);
-$modal-header-font-weight: var(--dt-font-heading-m-weight);
-$modal-header-padding: var(--dt-spacing-e);
-$modal-header-padding-block-end: var(--dt-spacing-d);
-
-$modal-contentWrapper-font-size: calc(var(--dt-font-size-16) * 1px);
-$modal-contentWrapper-line-height: calc(var(--dt-font-size-16-line-height) * 1px);
-$modal-contentWrapper-font-weight: var(--dt-font-weight-regular);
-$modal-contentWrapper-padding: var(--dt-spacing-e);
-$modal-contentWrapper-padding-block-start: var(--dt-spacing-a);
-
 .c-modal {
-  border-radius: $modal-border-radius;
-  border: none;
-  font-family: $modal-font;
-  background-color: $modal-bg-color;
-
-  padding: 0;
-  inline-size: $modal-size-m;
-
-  box-shadow: $modal-elevation;
-
-  & .c-modal-heading {
-    font-size: $modal-header-font-size;
-    line-height: $modal-header-font-line-height;
-    font-weight: $modal-header-font-weight;
-
-    margin: 0;
-    padding-block: $modal-header-padding $modal-header-padding-block-end;
-    padding-inline: $modal-header-padding;
-  }
-
-  & .c-modal-contentWrapper {
-    font-size: $modal-contentWrapper-font-size;
-    line-height: $modal-contentWrapper-line-height;
-    font-weight: $modal-contentWrapper-font-weight;
-
-    padding-block: $modal-contentWrapper-padding-block-start $modal-contentWrapper-padding;
-    padding-inline: $modal-contentWrapper-padding;
-  }
+    // Custom Property Declarations
+    // These are defined here instead of :host to encapsulate them inside Shadow DOM
+    --modal-size-s: 450px;
+    --modal-size-m: 600px;
+    --modal-size-l: 1080px;
+
+    // The base modal defaults
+    --modal-border-radius: var(--dt-radius-rounded-d);
+    --modal-font: var(--dt-font-interactive-m-family);
+    --modal-bg-color: var(--dt-color-container-default);
+    --modal-elevation: var(--dt-elevation-04);
+
+    border-radius: var(--modal-border-radius);
+    border: none;
+    font-family: var(--modal-font);
+    background-color: var(--modal-bg-color);
+
+    padding: 0;
+    inline-size: var(--modal-size-m);
+
+    box-shadow: var(--modal-elevation);
+
+    & .c-modal-heading {
+        // Modal header Custom Props
+        --modal-header-font-size: calc(var(--dt-font-heading-m-size--wide) * 1px);
```

Comment:
I'm curious, why the `* 1px`?

Reply:
Good question! So we store our sizes as unitless values (as we use `px` on web, iOS use `pt` – I'm not sure what Android uses).

You can't just add a 'px' string to this number, otherwise you get a type mismatch in the variable, so we have to multiply it by 1px to convert it to a pixel value.

We'll be looking at turning this into a shared function/mixin shortly when we pull together some useful shared styling for use across components.

Reply:
Nice one, I didn't even know that is a thing, mind => blown